integrators/ordinary_differential_equations.hpp

```diff
@@ -34,9 +34,137 @@ using base::Status;
 using geometry::Instant;
 using numerics::DoublePrecision;
 using quantities::Difference;
+using quantities::Quotient;
 using quantities::Time;
 using quantities::Variation;
 
+// A differential equation of the form y′ = f(y, t).
+// |State| is the type of y.
+template<typename... StateTypes>
+struct ExplicitFirstOrderOrdinaryDifferentialEquation final {
+  using State = std::tuple<std::vector<StateTypes>...>;
+  // NOTE(eggrobin): we cannot use |Variation| here because of a bug in MSVC
+  // when parameter packs interact with defaulted template parameters (and
+  // possibly decltype), but this is just
+  //   std::tuple<std::vector<Variation<StateTypes>>...>.
+  using StateVariation = std::tuple<
+      std::vector<Quotient<Difference<StateTypes, StateTypes>, Time>>...>;
+
+  using RightHandSideComputation =
+      std::function<Status(Instant const& t,
+                           State const& state,
+                           StateVariation& derivatives)>;
```

Comment:
Pick either "variation" or "derivative" and use only one consistently.

Reply:
Done.